### ISSUE TITLE
Failure in test_ncut_stable_subgraph on Python 3.6

### ISSUE THREAD

**Issue by matthew-brett** [scikit-image/scikit-image#2450]
Test failure for Python 3.6 (when testing current release wheel):

```
======================================================================
FAIL: Test to catch an error thrown when subgraph has all equal edges.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/venv/lib/python3.6/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/venv/lib/python3.6/site-packages/numpy/testing/decorators.py", line 147, in skipper_func
    return f(*args, **kwargs)
  File "/venv/lib/python3.6/site-packages/skimage/future/graph/tests/test_rag.py", line 180, in test_ncut_stable_subgraph
    assert new_labels.max() == 0
AssertionError
```

https://travis-ci.org/scikit-image/scikit-image-wheels/jobs/191712740

**Comment by JDWarner** [comment]
Could be an upstream bug; it looks like NetworkX doesn't test against 3.6 on Travis.  NetworkX uses dictionaries heavily and a more compact `dict` was a banner change in the release of Python 3.6.  Can't reproduce right now at work, but will look into it this weekend unless someone else beats me to the punch.

**Comment by matthew-brett** [comment]
I added a Python 3.6 entry for travis here:

https://github.com/scikit-image/scikit-image/pull/2451

It finds the same error for Python 3.6: https://travis-ci.org/matthew-brett/scikit-image/jobs/191828439#L3521

**Comment by matthew-brett** [comment]
See https://github.com/scikit-image/scikit-image/pull/2451#issuecomment-272932260 - I suspect this is not a Networkx bug, but a bug in our assumptions about the graph, and the graph has changed as a result of the change in Python 3.6 dict ordering.

**Comment by matthew-brett** [comment]
I think this is a blocker for the 0.13 release - see discussion at #2451 .

Enabling the Python 3.6 tests now just involves uncommenting the section in the `.travis.yml` file (now we've merged #2451).

**Comment by matthew-brett** [comment]
Any progress here?  It would be great to get out some Python 3.6 wheels, that pass the tests.

**Comment by soupault** [comment]
@jni Could you help us with the name of a person who is familiar with this part?

**Comment by jni** [comment]
Sorry for the delays! @vighneshbirodkar is the expert, but I'll have a look also.

**Comment by sciunto** [comment]
Networkx released a version 1.11 by the end of January. Does it solve the issue @matthew-brett? https://github.com/networkx/networkx/releases

**Comment by sciunto** [comment]
oops, wrong year... :(

**Comment by soupault** [comment]
@matthew-brett should be closed soon via #2511 .
